packages/flutter/lib/src/material/pickers/date_picker_dialog.dart

```diff
@@ -458,52 +481,70 @@ class _DatePickerDialogState extends State<_DatePickerDialog> {
-                  children: <Widget>[
-                    header,
-                    Expanded(child: picker),
-                    actions,
-                  ],
-                );
-              case Orientation.landscape:
-                return Row(
-                  mainAxisSize: MainAxisSize.min,
-                  crossAxisAlignment: CrossAxisAlignment.stretch,
-                  children: <Widget>[
-                    header,
-                    Flexible(
-                      child: Column(
+      child: Shortcuts(
+        shortcuts: _shortcutMap,
+        child: Actions(
+          actions: _actionMap,
+          child: Focus(
+            autofocus: dialogAutoFocus,
+            includeSemantics: false,
+            child: AnimatedContainer(
+              width: dialogSize.width,
+              height: dialogSize.height,
+              duration: _dialogSizeAnimationDuration,
+              curve: Curves.easeIn,
+              child: MediaQuery(
+                data: MediaQuery.of(context).copyWith(
+                  textScaleFactor: textScaleFactor,
+                ),
+                child: Builder(builder: (BuildContext context) {
+                  switch (orientation) {
+                    case Orientation.portrait:
+                      return Column(
                         mainAxisSize: MainAxisSize.min,
                         crossAxisAlignment: CrossAxisAlignment.stretch,
                         children: <Widget>[
+                          header,
                           Expanded(child: picker),
                           actions,
                         ],
-                      ),
-                    ),
-                  ],
-                );
-            }
-            return null;
-          }),
+                      );
+                    case Orientation.landscape:
+                      return Row(
+                        mainAxisSize: MainAxisSize.min,
+                        crossAxisAlignment: CrossAxisAlignment.stretch,
+                        children: <Widget>[
+                          header,
+                          Flexible(
+                            child: Column(
+                              mainAxisSize: MainAxisSize.min,
+                              crossAxisAlignment: CrossAxisAlignment.stretch,
+                              children: <Widget>[
+                                Expanded(child: picker),
+                                actions,
+                              ],
+                            ),
+                          ),
+                        ],
+                      );
+                  }
+                  return null;
+                }),
+              ),
+            ),
+          ),
         ),
       ),
       insetPadding: const EdgeInsets.symmetric(horizontal: 16.0, vertical: 24.0),
       clipBehavior: Clip.antiAlias,
     );
   }
 }
+
+class _WidgetActivateIntent extends Intent {
```

Comment:
I'm not super familiar with keyboard mappings/shortcuts. Is this some boilerplate that is necessary?

Reply:
This was a way to have a key shortcut activate a specific widget. I was doing this so that using the 'escape' key would look like you had tapped on the Cancel button and 'enter' would look like you tapped on the Ok button. I have decided there is a better way to handle this in a generic way for dismissal, and the confirm shortcut is a bit problematic. I will rework these in follow on PRs.